### PR TITLE
roadrunner: init at 2023.2.2

### PR DESCRIPTION
**Pull request by shyim**
###### Description of changes

I want to use Roadrunner for my projects. So packaged also for Nixpkgs

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — APPROVED** [review]
LGTM

`buildGoModule` doesn't support `finalAttrs` pattern, so it's all good as it is now for me.

File: pkgs/servers/roadrunner/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+, installShellFiles
+}:
+
+buildGoModule rec {
+  pname = "roadrunner";
+  version = "2023.1.0";
+  src = fetchFromGitHub {
+    repo = "roadrunner";
+    owner = "roadrunner-server";
+    rev = "v2023.1.0";
```
Comment (drupol) [inline pkgs/servers/roadrunner/default.nix:13]: Can you update this to the latest version so we can proceed ?

File: pkgs/servers/roadrunner/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    hash = "sha256-whstzXfk3gZ8KZ+q1XjjXxUh1jeQEeZbtrKPRKnvVZU=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  # One test asserts wrong tmp dir, @see https://github.com/roadrunner-server/roadrunner/issues/1545
```
Comment (drupol) [inline pkgs/servers/roadrunner/default.nix:21]: I see that this has been fixed, can we enable checks now ?

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
@shyim Do you think you could have a look at this PR at some point?

File: pkgs/servers/roadrunner/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      --replace "127.0.0.1:55555" "127.0.0.1:55554"
+
+    substituteInPlace internal/rpc/test/config_rpc_ok.yaml \
+      --replace "127.0.0.1:55555" "127.0.0.1:55554"
+
+    substituteInPlace internal/rpc/test/config_rpc_conn_err.yaml \
+      --replace "127.0.0.1:0" "127.0.0.1:55554"
+  '';
+
+  vendorSha256 = "sha256-pRZaJ1PBtshhna71V86IJ0VKs0u9wCFG27mghcE/8xY";
```
Comment (drupol) [inline pkgs/servers/roadrunner/default.nix:47]:
```suggestion
  vendorHash = "sha256-pRZaJ1PBtshhna71V86IJ0VKs0u9wCFG27mghcE/8xY";
```

File: pkgs/servers/roadrunner/default.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  meta = {
```
Comment (drupol) [inline pkgs/servers/roadrunner/default.nix:49]: Could you sort meta attributes alphabetically?

**Review by drupol — CHANGES_REQUESTED** [review]
2 minor changes and it's good to go!

**Review by drupol — APPROVED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
Crap,I forgot to ask you to add the `meta.mainProgram` attribute 🫤

**Comment by shyim** [comment]
Done :)

**Review by drupol — APPROVED** [review]
Thank you :)

**Pull request merged**